diff --git a/Sources/SwiftTerm/SelectionService.swift b/Sources/SwiftTerm/SelectionService.swift
index 14d6a24..e4e32b3 100644
--- a/Sources/SwiftTerm/SelectionService.swift
+++ b/Sources/SwiftTerm/SelectionService.swift
@@ -581,7 +581,7 @@ public class SelectionService: CustomDebugStringConvertible {
         case " ":
             includeFunc = { ch in ch == " " }
         case let ch where ch.isLetter || ch.isNumber:
-            includeFunc = { ch in ch.isLetter || ch.isNumber || ch == "." || ch == "_" || ch == "-" }
+            includeFunc = { ch in ch.isLetter || ch.isNumber || ch == "." || ch == "_" || ch == "-" || ch == "@" || ch == "+" || ch == "~" }
         default:
             return position
         }
@@ -631,7 +631,7 @@ public class SelectionService: CustomDebugStringConvertible {
             // Select all white space
             simpleScanSelection (from: position, in: buffer) { ch in ch == " " }
         case let ch where ch.isLetter || ch.isNumber:
-            simpleScanSelection (from: position, in: buffer) { ch in ch.isLetter || ch.isNumber || ch == "." || ch == "_" || ch == "-" }
+            simpleScanSelection (from: position, in: buffer) { ch in ch.isLetter || ch.isNumber || ch == "." || ch == "_" || ch == "-" || ch == "@" || ch == "+" || ch == "~" }
         case "{":
             fallthrough
         case "(":
